README.md

```diff
@@ -1,3 +1,148 @@
+
+[Sopel]: https://pypi.org/project/sopel/
+[General Category]: https://en.wikipedia.org/wiki/Unicode_character_property#General_Category
+[`unicodedata2`]: https://pypi.org/project/unicodedata2/
+[`unicode_age`]: https://pypi.org/project/unicode-age/
+
+## Installing
+
+Releases are hosted on PyPI, so after installing Sopel, all you need is `pip`:
+
+```shell
+$ pip install sopel-unicode
+```
+
+### Optional features
+
+`sopel-unicode` is designed to act as a drop-in replacement for the original built-in plugin on a basic installation.
+To enable all the optional features:
+
+```shell
+$ pip install sopel-unicode[all]
+```
+<!-- TODO:SnoopJ list and explain options for granular extras once CLDR is included -->
+
+## Usage
+
+Note that output given in this section corresponds to `sopel-unicode[all]` except where noted. Output layout may differ
+if some optional dependencies are missing.
+
+### Codepoint lookup
+
+The `unicode` (short-form `u`) command provides lookup of codepoints in a provided string. Input characters defined by
+the configuration option `ignore_chars` are ignored.
+
+Lookup uses [`unicodedata2`] if it is available, and falls back on [stdlib `unicodedata`] otherwise.
+
+[stdlib `unicodedata`]: https://docs.python.org/3/library/unicodedata.html
+
+```
+< SnoopJ> !unicode 🫩
+< terribot> [unicode] (🫩): U+1FAE9 v16.0 (So) FACE WITH BAGS UNDER EYES
+< SnoopJ> !u 🏴☠ 
+< terribot> [unicode] (🏴): U+1F3F4 v7.0 (So) WAVING BLACK FLAG
+< terribot> [unicode] (‍): U+200D v1.1 (Cf) ZERO WIDTH JOINER
+< terribot> [unicode] (☠): U+2620 v1.1 (So) SKULL AND CROSSBONES
+```
+
+It is sometimes convenient to discard all ASCII characters from lookup, which can be done with the
+`unicode:noascii`(`u:noascii`) command:
+
+```
+< SnoopJ> !u:noascii ça va?
+< terribot> [unicode] (ç): U+00E7 v1.1 (Ll) LATIN SMALL LETTER C WITH CEDILLA
+```
+
+The `unicode:raw` (`u:raw`) command is provided to avoid discarding *any* codepoints when performing lookup.
+
+```
+< SnoopJ> !unicode:raw a b
+< terribot> [unicode] (a): U+0061 v1.1 (Ll) LATIN SMALL LETTER A
+< terribot> [unicode] ( ): U+0020 v1.1 (Zs) SPACE
+< terribot> [unicode] (b): U+0062 v1.1 (Ll) LATIN SMALL LETTER B
+```
+
+Individual codepoints can also be looked up with hex notation, in either `U+NNNN` form, `0xNNNN` form, or `\uNNNN` form.
+
+```
+< SnoopJ> !unicode U+037E
+< terribot> [unicode] (;): U+037E v1.1 (Po) GREEK QUESTION MARK
+< SnoopJ> !u 0xBEEF
+< terribot> [unicode] (뻯): U+BEEF v2.0 (Lo) HANGUL SYLLABLE BBEGS
+< SnoopJ> !u \u732b
+< terribot> [unicode] (猫): U+732B v1.1 (Lo) CJK UNIFIED IDEOGRAPH-732B
+```
+
+Note that the `\u` notation is *not restricted* in the same way as the same notation for Python literals. You may use as
+many or as few hex digits as you like.
+
+```
+< SnoopJ> !u \1
+< terribot> [unicode] (): U+0001 v1.1 (Cc) START OF HEADING
+< SnoopJ> !u \12345
+< terribot> [unicode] (𒍅): U+12345 v5.0 (Lo) CUNEIFORM SIGN URU TIMES KI
```

Comment:
Just `\` with no `u` is supposed to work? That's a little weird, but fine if intended. At first glance it looks like a typo in the readme.

Reply:
Nope, that's a mistake.